Source/Core/VideoBackends/Vulkan/TextureCache.h

```diff
@@ -54,34 +31,40 @@ class TextureCache : public TextureCacheBase
   bool CompileShaders() override;
   void DeleteShaders() override;
 
-  TCacheEntryBase* CreateTexture(const TCacheEntryConfig& config) override;
+  std::unique_ptr<AbstractTexture> CreateTexture(const TextureConfig& config) override;
 
-  void ConvertTexture(TCacheEntryBase* base_entry, TCacheEntryBase* base_unconverted, void* palette,
+  void ConvertTexture(TCacheEntry* destination, TCacheEntry* source, void* palette,
                       TlutFormat format) override;
 
   void CopyEFB(u8* dst, const EFBCopyFormat& format, u32 native_width, u32 bytes_per_row,
                u32 num_blocks_y, u32 memory_stride, bool is_depth_copy,
                const EFBRectangle& src_rect, bool scale_by_half) override;
 
-  void CopyRectangleFromTexture(TCacheEntry* dst_texture, const MathUtil::Rectangle<int>& dst_rect,
-                                Texture2D* src_texture, const MathUtil::Rectangle<int>& src_rect);
+  void CopyRectangleFromTexture(const VKTexture* dst_texture,
+                                const MathUtil::Rectangle<int>& dst_rect, Texture2D* src_texture,
+                                const MathUtil::Rectangle<int>& src_rect);
 
   bool SupportsGPUTextureDecode(TextureFormat format, TlutFormat palette_format) override;
 
-  void DecodeTextureOnGPU(TCacheEntryBase* entry, u32 dst_level, const u8* data, size_t data_size,
+  void DecodeTextureOnGPU(TCacheEntry* entry, u32 dst_level, const u8* data, size_t data_size,
                           TextureFormat format, u32 width, u32 height, u32 aligned_width,
                           u32 aligned_height, u32 row_stride, const u8* palette,
                           TlutFormat palette_format) override;
 
+  VkShaderModule GetEFBColorToTexShader() const;
```

Comment:
See my point below, I think the implementation is incorrect here. RenderPass should probably be renamed to "TextureCopyRenderPass", as there are many render passes in the backend. The raw suffix can be dropped from TextureUploadBuffer, since streambuffer is not abstracted in common.